erpnext/stock/doctype/item/item.py

```diff
@@ -985,6 +982,22 @@ def validate_auto_reorder_enabled_in_stock_settings(self):
 				)
 
 
+def convert_erpnext_to_barcodenumber(erpnext_number):
+	convert = {
+		"UPC-A": "UPCA",
+		"CODE-39": "CODE39",
+		"EAN": "EAN13",
+		"EAN-12": "EAN",
+		"EAN-8": "EAN8",
```

Comment:
This is somewhat problematic, earlier we allowed all different EAN types to be considered as just "EAN"


reference  https://github.com/frappe/erpnext/issues/34233

Reply:
Let's relax it. The point of this validation is to avoid errors, they shouldn't become a nuisance :)

Reply:
@ankush: Should we implement the behaviour as before? (Allow all different EAN types to be considered as just "EAN")
If yes, I can do that. I was not clear about the previous behavior.

Reply:
@developsessions yeah that would be better for existing users. Otherwise resaving item will start failing.

Reply:
@ankush you're right. I will prepare the change and create an pull request for this fix.